### PR TITLE
#23338 Fix `Restore Defaults` for Metadata preference page

### LINKED ISSUE #23338: The Restore Defaults doesn't work for the Metadata read setting
### Description

The Restore Defaults button doesn't work for the open separate connection for metadata read setting

### DBeaver Version

24.0.2

### Operating System

Win10

### Database and driver

_No response_

### Steps to reproduce

1. go to Preferences > Connections > Metadata
2. change  "open separate connection for metadata read" setting and press The Restore Defaults button

AR: no visible changes

https://github.com/dbeaver/dbeaver/assets/112612526/bf314368-f759-4d67-a0e1-1504151c5e4b

ER: a default value is set for the setting

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageMetaData.java [hunk 1]
```diff
@@ -158,7 +158,9 @@ protected void clearPreferences(DBPPreferenceStore store)
     protected void performDefaults() {
         DBPPreferenceStore store = DBWorkbench.getPlatform().getPreferenceStore();
         readExpensiveCheck.setSelection(store.getDefaultBoolean(ModelPreferences.READ_EXPENSIVE_PROPERTIES));
-        UIUtils.setComboSelection(separateMetaConnectionCombo, store.getDefaultString(ModelPreferences.META_SEPARATE_CONNECTION));
+        separateMetaConnectionCombo.select(metaUseSeparateConnectionValues.indexOf(
+            SeparateConnectionBehavior.parse(store.getString(ModelPreferences.META_SEPARATE_CONNECTION))
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageMetaData.java]: Why not `getDefaultString`?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageMetaData.java]: yes

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageMetaData.java]: Wat?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageMetaData.java]: just yes :D
(it's a mistake)

**Review by ShadelessFox — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

**Pull request merged**